### PR TITLE
refactor(sbb-toast): lit migration

### PR DESCRIPTION
**Pull request by TomMenga**
## Preflight Checklist

- [x] I have read the [Contributing Guidelines](https://github.com/lyne-design-system/lyne-components/blob/master/CONTRIBUTING.md) for this project.
- [x] I agree to follow the [Code of Conduct](https://github.com/lyne-design-system/lyne-components/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
- [x] I have searched the [pull request tracker](https://github.com/lyne-design-system/lyne-components/pulls) for a Pull Request (PR) that matches the one I want to submit, without success.

## Issue

This PR Closes #

## Pull request checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)

See [Review Guidelines](../REVIEW.md) for more information what is checked during review process.

## Changes

Changes in this pull request:

- Sbb-toast lit migration

## Browsers

I tested the build on the following browsers:

- [ ] Firefox Desktop
- [ ] Chrome Desktop
- [ ] Edge Desktop
- [ ] Safari Desktop
- [ ] Chrome Mobile
- [ ] Safari Mobile

## Screen readers

I tested the build on the following browsers:

- [ ] JAWS Firefox Desktop
- [ ] JAWS Chrome Desktop
- [ ] NVDA Firefox Desktop
- [ ] NVDA Chrome Desktop
- [ ] VoiceOver Safari Desktop
- [ ] VoiceOver Chrome Desktop
- [ ] VoiceOver Safari Mobile
- [ ] Android Accessibility Suite Chrome Mobile

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/components/sbb-toast/sbb-toast.tsx [hunk 1]
```diff
@@ -21,98 +6,108 @@ import {
...
+import '../sbb-link';
+import '../sbb-button';
 
 // A global collection of existing toasts
-const toastRefs = new Set<HTMLSbbToastElement>();
+const toastRefs = new Set<SbbToast>();
 
 /**
  * @slot unnamed - Use this to document a slot.
  */
```
Comment (kyubisation) [inline src/components/sbb-toast/sbb-toast.tsx:27]:
```suggestion
```

File: src/components/sbb-toast/sbb-toast.tsx [hunk 2]
```diff
@@ -21,98 +6,108 @@ import {
...
-})
-export class SbbToast implements ComponentInterface {
+export const events = {
+  willOpen: 'will-open',
+  didOpen: 'did-open',
+  willClose: 'will-close',
+  didClose: 'did-close',
+};
+
+@customElement('sbb-toast')
```
Comment (kyubisation) [inline src/components/sbb-toast/sbb-toast.tsx:35]:
```suggestion
/**
 * @slot unnamed - Use this to document a slot.
 */
@customElement('sbb-toast')
```

File: src/components/sbb-toast/sbb-toast.tsx [hunk 3]
```diff
@@ -251,45 +252,55 @@ export class SbbToast implements ComponentInterface {
...
+    setAttribute(this, 'data-has-action', this._namedSlots['action'] || this.dismissible);
+    // ####
+
+    return html`
+      <div class="sbb-toast__overlay-container">
+        <div
+          class="sbb-toast"
+          ${/* Firefox needs 'role' to enable screen readers */ ''}
+          role=${this._role ?? nothing}
+          @animationend=${(event: AnimationEvent) => this._onToastAnimationEnd(event)}
```
Comment (kyubisation) [inline src/components/sbb-toast/sbb-toast.tsx:268]: Nitpick (https://lit.dev/docs/templates/expressions/#event-listener-expressions)
```suggestion
          @animationend=${this._onToastAnimationEnd}
```

File: src/components/sbb-toast/sbb-toast.tsx [hunk 4]
```diff
@@ -251,45 +252,55 @@ export class SbbToast implements ComponentInterface {
...
+        >
+          <div class="sbb-toast__icon">
+            <slot name="icon">
+              ${this.iconName ? html`<sbb-icon name=${this.iconName} />` : nothing}
+            </slot>
+          </div>
+
+          <div class="sbb-toast__content" aria-live=${this.politeness}>
+            <slot @slotchange=${(event) => this._onContentSlotChange(event)}></slot>
```
Comment (kyubisation) [inline src/components/sbb-toast/sbb-toast.tsx:277]: Nitpick (https://lit.dev/docs/templates/expressions/#event-listener-expressions)
```suggestion
            <slot @slotchange=${this._onContentSlotChange}></slot>
```

File: src/components/sbb-toast/sbb-toast.tsx [hunk 5]
```diff
@@ -251,45 +252,55 @@ export class SbbToast implements ComponentInterface {
...
+          </div>
+
+          <div class="sbb-toast__action">
+            <slot name="action" @slotchange=${(event) => this._onActionSlotChange(event)}>
```
Comment (kyubisation) [inline src/components/sbb-toast/sbb-toast.tsx:281]: Nitpick (https://lit.dev/docs/templates/expressions/#event-listener-expressions)
```suggestion
            <slot name="action" @slotchange=${this._onActionSlotChange}>
```

**Review by kyubisation — APPROVED** [review]
LGTM 👍 
Some improvement suggestions

File: src/components/sbb-toast/sbb-toast.tsx (hunk 3 above)
Comment (TomMenga) [reply src/components/sbb-toast/sbb-toast.tsx:268]: It makes sense 👍 I just wanna be sure that the right context is bonded

**Pull request merged**